lib/solr/core_configuration/core.rb

```diff
@@ -0,0 +1,34 @@
+module Solr
+  module CoreConfiguration
+    class Core
+      attr_reader :name, :url, :fields
+
+      def initialize(name:, url:, fields:)
+        @name = name
```

Comment:
do we need both name and url? Isn't a core only defined by a name and then we can append it to the base URL?

Reply:
A core can be initialized with name and/or URL. For example, when a developer uses just SOLR_URL, the core doesn't have a name:
```ruby
# core doesn't have its name because it's included in the URL
Solr.configure do |config|
  config.url = 'http://localhost:8983/solr/test-core' # URL with core's name
end
```

Reply:
core name is `test-core` in this case. In both cases core is identified by a path segment.

Reply:
@Valve, I already tried to find solr's URL schema, but without luck :(
Do you know where I can read more about it? That would be very useful.
For example, I can imagine several URLs:
```
http://localhost:8983/solr/test-core
http://localhost:8983/test-core
http://localhost:8983/solr
```
How can I be sure if path is a core name or not?